### ISSUE TITLE
Instantiating a recursive procedure gets into infinite recursion

### ISSUE THREAD

**Issue by radl97** [leventeBajczi/theta#1]
stack overflow error:
```
	at hu.bme.mit.theta.xcfa.dsl.XcfaStatement.instantiate(XcfaStatement.java:59)
	at hu.bme.mit.theta.xcfa.dsl.XcfaEdge.lambda$instantiate$1(XcfaEdge.java:39)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at hu.bme.mit.theta.xcfa.dsl.XcfaEdge.instantiate(XcfaEdge.java:39)
	at hu.bme.mit.theta.xcfa.dsl.XcfaProcedureSymbol.lambda$instantiate$3(XcfaProcedureSymbol.java:100)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at hu.bme.mit.theta.xcfa.dsl.XcfaProcedureSymbol.instantiate(XcfaProcedureSymbol.java:100)
	at hu.bme.mit.theta.xcfa.dsl.XcfaProcedureSymbol.instantiate(XcfaProcedureSymbol.java:35)
	at hu.bme.mit.theta.xcfa.dsl.XcfaStatement$StmtCreatorVisitor.visitProcCallStmt(XcfaStatement.java:131)
	at hu.bme.mit.theta.xcfa.dsl.XcfaStatement$StmtCreatorVisitor.visitProcCallStmt(XcfaStatement.java:67)
	at hu.bme.mit.theta.xcfa.dsl.gen.XcfaDslParser$ProcCallStmtContext.accept(XcfaDslParser.java:3993)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:46)
	at hu.bme.mit.theta.xcfa.dsl.gen.XcfaDslBaseVisitor.visitStmt(XcfaDslBaseVisitor.java:349)
	at hu.bme.mit.theta.xcfa.dsl.gen.XcfaDslParser$StmtContext.accept(XcfaDslParser.java:3457)
	at hu.bme.mit.theta.xcfa.dsl.XcfaStatement.instantiate(XcfaStatement.java:59)
```

The problem seems to be in XcfaProcedureSymbol:
https://github.com/leventeBajczi/theta/blob/a1c2a64503451510bdac3734247024ac92feafcd/subprojects/xcfa/src/main/java/hu/bme/mit/theta/xcfa/dsl/XcfaProcedureSymbol.java#L88
This instantiate function has a return value caching that, if once ran, every subsequent run returns the already built Procedure.
So first procedure==null, then when entered, it tries to build the structure for procedure. When it ended, procedure gets its value.
The problem is that *while still building the procedure itself*, it's called again, when it's needed by a call within the procedure (recursive function).

The point is: you cannot use builder for a procedure that needs to reference itself

**Comment by leventeBajczi** [comment]
Hopefully solved in commit `33a016a`. Please re-test.

**Comment by radl97** [comment]
Re-tested. With minor fixes, it is now working ^_^
